extension/syntaxes/templateSettingsSyntax.json

```diff
@@ -0,0 +1,207 @@
+{
+  "$id": "NAB AL Tools Template Settings Syntax",
+  "title": "Root",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$comment": "Created with https://jsonformatter.org/json-to-jsonschema",
+  "definitions": {},
+  "type": "object",
+  "required": [
+    "mappings"
+  ],
+  "description": "Specifies the settings for NAB AL Tools Template Conversion",
+  "properties": {
+    "mappings": {
+      "$id": "#root/mappings",
+      "title": "Mappings",
+      "description": "Specifies all settings for how texts should be replaced and files renamed.",
+      "type": "array",
+      "default": [],
+      "items": {
+        "$id": "#root/mappings/items",
+        "title": "Items",
+        "type": "object",
+        "required": [
+          "description",
+          "example"
+        ],
+        "properties": {
+          "description": {
+            "$id": "#root/mappings/items/description",
+            "title": "Description",
+            "description": "Specifies a description of the current setting. This will be shown for the user.",
+            "type": "string",
+            "default": "",
+            "examples": [],
+            "pattern": "^.*$"
+          },
+          "example": {
+            "$id": "#root/mappings/items/example",
+            "title": "Example",
+            "description": "Specifies an example of what value should be supplied by the user. This will be shown for the user.",
+            "type": "string",
+            "default": "",
+            "examples": [],
+            "pattern": "^.*$"
+          },
+          "default": {
+            "$id": "#root/mappings/items/default",
+            "title": "Default",
+            "description": "Specifies a default value as a suggestion for the user. Use the special value $(guid) to get a new GUID as default value. This will be shown for the user, if supplied.",
+            "type": "string",
+            "default": "",
+            "examples": [],
+            "pattern": "^.*$"
+          },
+          "renameFile": {
```

Comment:
Plural perhaps?

Reply:
Yes, `renameFiles` is better.

Can you help me with a new name on `searchAndReplace`? 

Some ideas:
- `replacements`?
- `substitutions`?

Reply:
I like substitutions. Just spitballing:
- `stringSubstitutions`
- `placeholderSubstitutions`
- `placeholderReplacements`

I think `placeholderReplacements` is the best one I came up with so far.

Reply:
We'll go for `placeholderSubstitutions` for now, it is still preview - we can change later if we want to! 😜